web/src/components/grouped-score-badge.tsx

```diff
@@ -5,88 +5,161 @@ import {
-    <div className={cn("text-xs", props.className)}>
-      {props.scores.map((s, i) => (
-        <span key={i} className="group/score ml-1 first:ml-0">
-          {isCategoricalDataType(s.dataType) || isBooleanDataType(s.dataType)
-            ? s.stringValue
-            : s.value?.toFixed(2)}
-          {s.comment && (
-            <HoverCard>
-              <HoverCardTrigger className="ml-1 inline-block cursor-pointer">
-                <MessageCircleMore size={12} />
-              </HoverCardTrigger>
-              <HoverCardContent className="overflow-hidden whitespace-normal break-normal">
-                <p>{s.comment}</p>
-              </HoverCardContent>
-            </HoverCard>
-          )}
-          <span className="group-last/score:hidden">,</span>
-        </span>
-      ))}
-    </div>
-  );
+const DETAIL_HOVER_AFTER = 2;
 
-  if (variant === "headings")
+export const ScoresAggregateCell = ({
+  aggregate,
+  showSingleValue = false,
+}: {
+  aggregate: QualitativeAggregate | QuantitativeAggregate;
+  showSingleValue?: boolean;
+}) => {
+  if (showSingleValue && aggregate.values.length === 1) {
     return (
-      <div className="flex items-center gap-3 text-nowrap">
-        {Object.entries(groupedScores)
-          .sort(([a], [b]) => (a < b ? -1 : 1))
-          .map(([name, scores]) => (
-            <div key={name}>
-              <div className="text-xs text-muted-foreground">{name}</div>
-              <ScoresOfGroup scores={scores} />
-            </div>
-          ))}
-      </div>
+      <SingleScoreValue
+        value={
+          aggregate.type === "QUALITATIVE"
+            ? aggregate.values[0]
+            : aggregate.average.toFixed(2)
+        }
+        comment={aggregate.comment}
+        showColorCoding
+      />
+    );
+  }
+
+  if (aggregate.type === "QUANTITATIVE") {
+    return (
+      <SingleScoreValue
+        value={`Ø ${aggregate.average.toFixed(2)}`}
+        showColorCoding
+      />
     );
-  else
+  } else if (aggregate.type === "QUALITATIVE") {
     return (
-      <>
-        {Object.entries(groupedScores)
-          .sort(([a], [b]) => (a < b ? -1 : 1))
-          .map(([name, scores]) => (
-            <Badge
-              variant="outline"
-              key={name}
-              className="grid grid-cols-[1fr,auto] gap-1 font-normal"
-            >
-              <p className="truncate" title={name}>
-                {name}:
-              </p>
-              <ScoresOfGroup scores={scores} />
-            </Badge>
-          ))}
-      </>
+      <div className="group">
+        {aggregate.distribution.length > DETAIL_HOVER_AFTER ? (
+          <HoverCard>
+            <HoverCardTrigger>
+              <div className="flex cursor-pointer flex-col group-hover:text-accent-dark-blue/55">
+                {aggregate.distribution
+                  .slice(0, DETAIL_HOVER_AFTER)
+                  .map(({ value, count }) => (
+                    <span key={value} className="truncate">
```

Comment:
truncate is nice, did not know about this and applied the classes one by one until now

issue here: we truncate the text including the value and the count, i'd move this out into a single component as it is used 3 times here, and then use a flex with val, count, where val is truncated while count is not